Add support for job time to live

balbirthomas opened the pull request:
Failure to remove resources used by completed jobs/pods can lead to resource exhaustion in the Kubernetes cluster. This commit adds support for setting a ttlSecondsAfterFinished parameter in the Kubernetes Job sepc. This parameter specifies the amount of time (in seconds) for which the resources used by completed job/pod will persist after it which it will be garbage collected by the Kube control manager.

The newly added parameter is optional and only set if it passed in through the `config` argument to the `make_job_spec` method.

## Purpose of PR

## Parts of the app this will impact

## Todos

## Additional information

## Related links

Commit:
Add support for job time to live

Failure to remove resources used by completed jobs/pods can lead to
resource exhaustion in the Kubernetes cluster. This commit adds support
for setting a ttlSecondsAfterFinished parameter in the Kubernetes Job
specification. This parameter specifies the amount of time (in seconds)
for which the resources used by completed job/pod will persist after it
which it will be garbage collected by the Kube control manager.

The newly added parameter is optional and only set if it passed in
through the `config` argument to the `make_job_spec` method.

gchazot commented on kubetools/kubernetes/config/job.py:
```diff
@@ -99,3 +99,8 @@ def make_job_config(
         },
     }
+
+    if config.get('ttlSecondsAfterFinished'):
```
`if 'ttlSecondsAfterFinished' in config`:

balbirthomas replied:
Ack. Will change.

balbirthomas replied:
Changes made and set for review.

gchazot reviewed (commented):
Please check if tests can be added for this new flag, and if any documentation/samples needs updated

balbirthomas commented:
> Please check if tests can be added for this new flag, and if any documentation/samples needs updated

Ack. Will do and ping you for review.

Commit:
Unpin flake8 version

This commit unpins flake8 version to make it compatible with the
current importib version.

Commit:
Added unit test for job config ttl

This commit adds a unit test to ensure the time to live option
for job configuration is set in the Kubernetes job spec, if
provided.

gchazot reviewed (approved):
LGTM.
Please be aware that, even when this is released into a new version of kubetools, it will still [not work on clusters running k8s < 1.21](https://github.com/EDITD/kubetools/issues/130).

One solution may be to backport this onto a branch based on `kubetools 13.8.1` and release that as a 13.8.x version

Commit:
Updated change log with Job TTL
